File: minesweeper/java-minesweeper/src/test/java/it/chicio/minesweeper/field/parser/FieldRowContentValuesChainParserTest.java
package it.chicio.minesweeper.field.parser;

import it.chicio.minesweeper.field.Field;
import org.junit.Before;
import org.junit.Test;

import static org.hamcrest.CoreMatchers.equalTo;
import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

public class FieldRowContentValuesChainParserTest {
    private FieldRowContentValuesChainParser fieldRowContentValuesChainParser;

    @Before
    public void setUp() {
        fieldRowContentValuesChainParser = new FieldRowContentValuesChainParser();
    }

    @Test
    public void canParseValidRow() {
        assertThat(fieldRowContentValuesChainParser.canParse("* . . *"), is(true));
    }

    @Test
    public void canNotParseOtherRows() {
        assertThat(fieldRowContentValuesChainParser.canParse("1 1"), is(false));
        assertThat(fieldRowContentValuesChainParser.canParse("0 0"), is(false));
    }

    @Test
    public void parseRow() {
        FieldsParsingStatus fieldsParsingStatus = fieldRowContentValuesChainParser.parse(new FieldsParsingStatusBuilder()
                .withCurrentField(new Field(new String[][]{{"*", "."}, {"", ""}}))
                .withCurrentRow(1)
                .withCurrentRowContent("* *")
                .build()
        );

        assertThat(fieldsParsingStatus.currentField.get(1, 0), is(equalTo("*")));
        assertThat(fieldsParsingStatus.currentField.get(1, 1), is(equalTo("*")));
    }

    @Test(expected = RuntimeException.class)
    public void failParseRow() {
        fieldRowContentValuesChainParser.parse(new FieldsParsingStatusBuilder()
                .withCurrentField(new Field(new String[][]{{"*", "."}, {"", ""}}))
                .withCurrentRow(1)
                .withCurrentRowContent("* * *")
                .build()
        );
    }
}

File: minesweeper/kotlin-mockk-minesweeper/src/test/kotlin/it/chicio/minesweeper/field/parser/FieldRowContentValuesChainParserTest.kt
package it.chicio.minesweeper.field.parser

import it.chicio.minesweeper.FieldFactory
import it.chicio.minesweeper.FieldsParsingStatusBuilder
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

@DisplayName("FieldRowContentValuesChainParser")
class FieldRowContentValuesChainParserTest {
    private lateinit var fieldRowContentValuesChainParser: FieldRowContentValuesChainParser

    @BeforeEach
    fun setUp() {
        fieldRowContentValuesChainParser = FieldRowContentValuesChainParser()
    }

    @Test
    fun `can parse valid row`() {
        assertTrue(fieldRowContentValuesChainParser.canParse("* . . *"))
    }

    @Test
    fun `cannot parse other rows`() {
        assertFalse(fieldRowContentValuesChainParser.canParse("1 1"))
        assertFalse(fieldRowContentValuesChainParser.canParse("0 0"))
    }

    @Test
    fun `parse row`() {
        val fieldsParsingStatus = fieldRowContentValuesChainParser.parse(FieldsParsingStatusBuilder(
                currentField = FieldFactory().make(arrayOf(arrayOf("*", "."), arrayOf("", ""))),
                currentRow = 1,
                currentRowContent = "* *"
        ).build())
        assertEquals(fieldsParsingStatus.currentField!![1, 0], "*")
        assertEquals(fieldsParsingStatus.currentField!![1, 1], "*")
    }

    @Test
    fun `fail parse row`() {
        assertThrows(RuntimeException::class.java) {
            fieldRowContentValuesChainParser.parse(FieldsParsingStatusBuilder(
                    currentField = FieldFactory().make(arrayOf(arrayOf("*", "."), arrayOf("", ""))),
                    currentRow = 1,
                    currentRowContent = "* * *"
            ).build())
        }
    }
}